src/bls.rs

```diff
@@ -1,15 +1,18 @@
-#[cfg(feature = "blst")]
 pub use blstrs::{
-    Bls12, Compress, Engine, Fp as Fq, Fp12 as Fq12, Fp2 as Fq2, FpRepr as FqRepr, G1Affine,
-    G1Compressed, G1Projective, G1Uncompressed, G2Affine, G2Compressed, G2Prepared, G2Projective,
-    G2Uncompressed, PairingCurveAffine, Scalar as Fr, ScalarRepr as FrRepr,
+    Bls12, Compress, Fp as Fq, Fp12 as Fq12, Fp2 as Fq2, G1Affine, G1Compressed, G1Projective,
+    G1Uncompressed, G2Affine, G2Compressed, G2Prepared, G2Projective, G2Uncompressed, Scalar as Fr,
 };
 
-#[cfg(feature = "pairing")]
-pub use paired::{
-    bls12_381::{
-        Bls12, Fq, Fq12, Fq2, FqRepr, Fr, FrRepr, G1Affine, G1Compressed, G1Uncompressed, G2Affine,
-        G2Compressed, G2Prepared, G2Uncompressed, G1 as G1Projective, G2 as G2Projective,
-    },
-    Compress, Engine, PairingCurveAffine,
-};
+use ff::PrimeField;
+use pairing::Engine;
+
+// Associates a `pairing::Engine` with its G1 base field; used to generate GPU field arithmetic for
+// that field.
+pub trait G1BaseField: Engine {
```

Comment:
Move this out, as a regular export, maybe calling it sth like `ExtendedEngine` and define the base fields for both `G1` and `G2`